packages/firestore/test/integration/api/database.test.ts

```diff
@@ -928,6 +928,58 @@ apiDescribe('Database', persistence => {
     });
   });
 
+  (persistence ? it : it.skip)(
+    'can clear persistence if the client has not been initialized',
+    async () => {
+      await withTestDoc(persistence, async docRef => {
+        await docRef.set({ foo: 'bar' });
+        const docSnap = await docRef.get({ source: 'cache' });
+        await expect(docSnap.data()).to.deep.equal({ foo: 'bar' });
+
+        const path = docRef.path;
+        const app = docRef.firestore.app;
+        const name = app.name;
+        const options = app.options;
+
+        // Verify data in cache persists after restart.
+        await app.delete();
+        const app2 = firebase.initializeApp(options, name);
+        const firestore2 = firebase.firestore!(app2);
+        await firestore2.enablePersistence();
+        const docRef2 = firestore2.doc(path);
+        const docSnap2 = await docRef2.get({ source: 'cache' });
+        expect(docSnap2.exists).to.be.true;
+        expect(docSnap2.data()).to.deep.equal({ foo: 'bar' });
+
+        // Verify data in cache is cleared if clearPersistence() is called.
+        await app2.delete();
+        await firestore2.clearPersistence();
+        const app3 = firebase.initializeApp(options, name);
+        const firestore3 = firebase.firestore!(app3);
+        await firestore3.enablePersistence();
+        const docRef3 = firestore3.doc(path);
+        await expect(
+          docRef3.get({ source: 'cache' })
+        ).to.eventually.be.rejectedWith(
+          'Failed to get document from cache. (However, this document may ' +
```

Comment:
Does this pass if you only pass in the first part of this error message?

Reply:
It's a direct string match passed in from FirestoreClient. 

I'm not sure how to partial string match when using `rejectedWith()`

Reply:
According to https://stackoverflow.com/questions/45227354/how-can-i-use-rejectedwith-to-perform-an-exact-match-of-the-error-message `chai` performs a substring match already.